gallery/visualtests/nxButton-2.js

```diff
@@ -0,0 +1,74 @@
+    it('has a dark background when hovered', hoverTest(primarySelector));
+    it('has a light blue background when clicked', clickTest(primarySelector));
+    it('has a white inner outline when focused', focusTest(primarySelector));
+    it('has a white inner outline and dark blue background when focused and hovered',
+        focusAndHoverTest(primarySelector));
+  });
+
+  describe('Small Secondary NxButton', function () {
+    const secondarySelector = `${selector} .nx-btn-bar:first-child .nx-btn--secondary.nx-btn--small`;
+
+    it('has a blue border and white background by default', simpleTest(secondarySelector));
+    it('has a blue background when hovered', hoverTest(secondarySelector));
+    it('has a blue border and light blue background when clicked', clickTest(secondarySelector));
+    it('has a blue border and darker blue inner outline when focused', focusTest(secondarySelector));
+    it('has a blue border, darker blue inner outline and blue background when focused and hovered',
+        focusAndHoverTest(secondarySelector));
+  });
+
+  describe('Small Tertiary NxButton', function () {
+    const tertiarySelector = `${selector} .nx-btn-bar:first-child .nx-btn--tertiary.nx-btn--small`;
+
+    it('has a grey border by default', simpleTest(tertiarySelector));
+    it('has a light indigo background when hovered', hoverTest(tertiarySelector));
+    it('has a light grey border and light indigo background when clicked', clickTest(tertiarySelector));
+    it('has a darker grey inner outline when focused', focusTest(tertiarySelector));
+    it('has a darker grey inner outline and light indigo background when focused and hovered',
+        focusAndHoverTest(tertiarySelector));
+  });
+
+  describe('Small Error NxButton', function () {
+    const errorSelector = `${selector} .nx-btn-bar:first-child .nx-btn--error.nx-btn--small`;
+
+    it('has a red background by default', simpleTest(errorSelector));
+    it('has a dark red background when hovered', hoverTest(errorSelector));
+    it('has a pink background when clicked', clickTest(errorSelector));
+    it('has a white inner outline when focused', focusTest(errorSelector));
+    it('has a dark red background and white inner outline when focused and hovered', focusAndHoverTest(errorSelector));
+  });
+
+  describe('Small Disabled NxButton', function () {
+    const disabledSelector =
+      '#nx-button-small-example .nx-btn-bar:first-child .nx-btn--small[disabled]';
+
+    it('has disabled styling', simpleTest(disabledSelector));
+    it('maintains disabled styling when hovered', hoverTest(disabledSelector));
+    it('maintains disabled styling when clicked', clickTest(disabledSelector));
+    it('maintains disabled styling when focused', focusTest(disabledSelector));
+  });
+
+  describe('Small Button with Icon', function () {
+    const withIconSelector = `${selector} .nx-btn-bar:nth-child(2) .nx-btn--small:has(span, svg.nx-icon)`;
```

Comment:
From MDN:
> The functional :has() CSS [pseudo-class](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes) represents an element if any of the [relative selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_selectors/Selector_structure#relative_selector) that are passed as an argument match at least one element when anchored against this element

So this'll match a button that contains _either_ a span or an svg.  Is that really what you want? Maybe it would be cleaner to just give this button an id.

Reply:
https://github.com/sonatype/sonatype-react-shared-components/pull/1220/commits/d7363ee2fb50ba3d6f4a2047de3470128540a8f3